src/passes/externalInputChecker/enumBoundChecker.ts

```diff
@@ -0,0 +1,67 @@
+import assert = require('assert');
+import { AST } from '../../ast/ast';
+import {
+  FunctionDefinition,
+  Identifier,
+  FunctionVisibility,
+  Statement,
+  ExpressionStatement,
+  VariableDeclaration,
+  FunctionCall,
+} from 'solc-typed-ast';
+import { ASTMapper } from '../../ast/mapper';
+
+export class EnumBoundChecker extends ASTMapper {
+  visitFunctionDefinition(node: FunctionDefinition, ast: AST): void {
+    if (FunctionVisibility.External === node.visibility) {
+      node.vParameters.vParameters.forEach((parameter) => {
+        if (parameter.typeString.slice(0, 4) === 'enum' && parameter.name !== undefined) {
+          const functionCall = this.generateFunctionCall(node, parameter, ast);
+          this.insertFunctionCall(node, functionCall, ast);
+        }
+      });
+    }
+    this.commonVisit(node, ast);
+  }
+
+  private generateFunctionCall(
+    node: FunctionDefinition,
+    parameter: VariableDeclaration,
+    ast: AST,
+  ): FunctionCall {
+    const enumStubArgument = new Identifier(
```

Comment:
(also applicable to all 3 files) There's a util function in nodeTemplates for this, `createIdentifier`

Reply:
Done